packages/http-signature-utils/src/utils/verification.ts

```diff
@@ -0,0 +1,120 @@
+import * as crypto from 'crypto'
+import { RequestLike } from 'http-message-signatures'
+import { verifyContentDigest } from 'httpbis-digest-headers'
+import { importJWK } from 'jose'
+import { JWK } from './jwk'
+
+export function validateHttpSigHeaders(request: RequestLike): boolean {
```

Comment:
I'm wondering if we should either just export a `verifyHttpSigHeaders` function or minimize overlap between that and `validateHttpSigHeaders`?
As is, I like how we're able to perform basic validation checks on expected headers in `auth`'s signature middleware before fetching the client key for actual signature verification. But I don't like how we call `validateHttpSigHeaders` again internally (with `getSigInputComponents`+`validateSigInputComponents`)

Reply:
I don't really know how we could improve on that here. Like you said, we want to validate the headers before fetching the key. When we validate the signature, however, we do need the components again. So the only option we have, I think, is to remove component validation from validating the signature because we always validate the headers first, which already includes validating the components.